homeassistant/components/filter/sensor.py

```diff
@@ -222,7 +226,10 @@ def _update_filter_sensor_state(self, new_state, update_ha=True):
         if self._icon is None:
             self._icon = new_state.attributes.get(ATTR_ICON, ICON)
 
-        if self._device_class is None:
+        if (
+            self._device_class is None
+            and new_state.attributes.get(ATTR_DEVICE_CLASS) in SENSOR_DEVICE_CLASSES
```

Comment:
We can store the device class attribute in a local variable so we don't need to do the lookup twice.

Reply:
That would need to happen before the `if` and would mean the lookup would happen **every** state update. 

`new_satate.attributes` is a dictionary, don't think it costs that much to query twice, and this does improve readability IMHO.

What do you say ?

Reply:
Ok. The alternative is to split the if check. But it's ok like this.